packages/odyssey-react-mui/src/labs/Switch.tsx

```diff
@@ -84,89 +87,120 @@ const StyledSwitchLabel = styled(FormLabel, {
       color: `${odysseyDesignTokens.TypographyColorDisabled} !important`,
     },
   }),
+
+  ...(isReadOnly && {
+    p: {
+      color: odysseyDesignTokens.HueNeutral700,
+    },
+    a: {
+      color: `${odysseyDesignTokens.HueNeutral700} !important`,
+    },
+  }),
 }));
 
 const SwitchTrack = styled("div", {
   shouldForwardProp: (prop) => !nonForwardedProps.includes(prop),
 })<
-  Pick<SwitchProps, "isChecked" | "isDisabled"> & {
+  Pick<SwitchProps, "isChecked" | "isDisabled" | "isReadOnly"> & {
     odysseyDesignTokens: DesignTokens;
   }
->(({ isChecked, isDisabled, odysseyDesignTokens }) => ({
-  position: "relative",
-  width: odysseyDesignTokens.Spacing7,
-  height: `calc(${odysseyDesignTokens.Spacing4} + ${odysseyDesignTokens.Spacing1})`,
-  borderRadius: odysseyDesignTokens.BorderRadiusOuter,
-  backgroundColor: isDisabled
-    ? odysseyDesignTokens.HueNeutral200
-    : isChecked
-      ? odysseyDesignTokens.PaletteSuccessLight
-      : odysseyDesignTokens.HueNeutral300,
-  transition: `background-color ${odysseyDesignTokens.TransitionDurationMain}`,
-}));
+>(({ isChecked, isDisabled, isReadOnly, odysseyDesignTokens }) => {
+  const getBackgroundColor = () => {
+    if (isDisabled) return odysseyDesignTokens.HueNeutral200;
```

Comment:
Can we do the traditional style if statement with brackets
```js
if (isDisabled) {
  return odysseyDesignTokens.HueNeutral200;
}

Reply:
switched this to CSS conditionals for simplicity